src/workerd/api/global-scope.h

```diff
@@ -331,12 +331,13 @@ class ServiceWorkerGlobalScope: public WorkerGlobalScope {
     return jsg::alloc<Navigator>();
   }
 
-  jsg::Unimplemented getOrigin() { return {}; }
+  kj::StringPtr getOrigin() { return ""_kj; }
```

Comment:
Does this have to return a string value in order for it to work? I'd much rather just remove it entirely so that `globalThis.origin` just always returns `undefined`.

It's *extremely unlikely* but there is a non-zero risk of this breaking folks if they happen to use this as any kind of detection mechanism, e.g.

```js
let isProbablyWorkers = false;
try {
  globalThis.origin;
} catch (err) {
  isProbablyWorkers = true;
}
```

Having this return `undefined` is probably safer than having it return an empty value.

Reply:
Done, incorporated. I hadn't appreciated the full spectrum of magic making the isolate available as an argument to `getOrigin()`.